### PR TITLE
WIP: run tagless-ember-components-codemod

### PR DESCRIPTION
**Pull request by jelhan**
This is the result of running the tagless-ember-components-codemod against Ember Bootstrap. It's more kind of keeping track of the changes needed to the codemod then meant to be merged.

The following open pull request have been merged in before:

- Support native classes: https://github.com/ember-codemods/tagless-ember-components-codemod/pull/44
- Support component with event handlers: https://github.com/ember-codemods/tagless-ember-components-codemod/pull/52
- Support aria role for native classes: https://github.com/ember-codemods/tagless-ember-components-codemod/pull/56

Steps:

```sh
# codemod does not support dynamical look up of template location yet
# renaming the folder to match the assumptions made by codemod
mv addon/templates/components/common addon/templates/components/base

# run codemod from branch that includes mentioned pull requests
npx jelhan/tagless-ember-components-codemod#merged-native-class-support

# revert the renaming
mv addon/templates/components/base addon/templates/components/common
```

Log:

```
npx: Installierte 377 in 13.735s
 🔍  Searching for component files...
app/components/bs-accordion.js: Could not find template at app/templates/components/bs-accordion.hbs
app/components/bs-alert.js: Could not find template at app/templates/components/bs-alert.hbs
app/components/bs-button-group.js: Could not find template at app/templates/components/bs-button-group.hbs
app/components/bs-button.js: Could not find template at app/templates/components/bs-button.hbs
app/components/bs-carousel.js: Could not find template at app/templates/components/bs-carousel.hbs
app/components/bs-collapse.js: Could not find template at app/templates/components/bs-collapse.hbs
app/components/bs-dropdown.js: Could not find template at app/templates/components/bs-dropdown.hbs
app/components/bs-form.js: Could not find template at app/templates/components/bs-form.hbs
app/components/bs-modal-simple.js: Could not find template at app/templates/components/bs-modal-simple.hbs
app/components/bs-modal.js: Could not find template at app/templates/components/bs-modal.hbs
app/components/bs-nav.js: Could not find template at app/templates/components/bs-nav.hbs
app/components/bs-navbar.js: Could not find template at app/templates/components/bs-navbar.hbs
app/components/bs-popover.js: Could not find template at app/templates/components/bs-popover.hbs
app/components/bs-progress.js: Could not find template at app/templates/components/bs-progress.hbs
app/components/bs-tab.js: Could not find template at app/templates/components/bs-tab.hbs
app/components/bs-tooltip.js: Could not find template at app/templates/components/bs-tooltip.hbs
app/components/bs-accordion/item.js: Could not find template at app/templates/components/bs-accordion/item.hbs
app/components/bs-dropdown/button.js: Could not find template at app/templates/components/bs-dropdown/button.hbs
app/components/bs-dropdown/menu.js: Could not find template at app/templates/components/bs-dropdown/menu.hbs
app/components/bs-dropdown/toggle.js: Could not find template at app/templates/components/bs-dropdown/toggle.hbs
app/components/bs-carousel/slide.js: Could not find template at app/templates/components/bs-carousel/slide.hbs
app/components/bs-button-group/button.js: Could not find template at app/templates/components/bs-button-group/button.hbs
app/components/bs-form/element.js: Could not find template at app/templates/components/bs-form/element.hbs
app/components/bs-form/group.js: Could not find template at app/templates/components/bs-form/group.hbs
app/components/bs-modal/body.js: Could not find template at app/templates/components/bs-modal/body.hbs
app/components/bs-modal/dialog.js: Could not find template at app/templates/components/bs-modal/dialog.hbs
app/components/bs-modal/footer.js: Could not find template at app/templates/components/bs-modal/footer.hbs
app/components/bs-modal/header.js: Could not find template at app/templates/components/bs-modal/header.hbs
app/components/bs-nav/item.js: Could not find template at app/templates/components/bs-nav/item.hbs
app/components/bs-nav/link-to.js: Could not find template at app/templates/components/bs-nav/link-to.hbs
app/components/bs-navbar/content.js: Could not find template at app/templates/components/bs-navbar/content.hbs
app/components/bs-navbar/link-to.js: Could not find template at app/templates/components/bs-navbar/link-to.hbs
app/components/bs-navbar/nav.js: Could not find template at app/templates/components/bs-navbar/nav.hbs
app/components/bs-navbar/toggle.js: Could not find template at app/templates/components/bs-navbar/toggle.hbs
app/components/bs-popover/element.js: Could not find template at app/templates/components/bs-popover/element.hbs
app/components/bs-progress/bar.js: Could not find template at app/templates/components/bs-progress/bar.hbs
app/components/bs-tab/pane.js: Could not find template at app/templates/components/bs-tab/pane.hbs
app/components/bs-tooltip/element.js: Could not find template at app/templates/components/bs-tooltip/element.hbs
app/components/bs-accordion/item/body.js: Could not find template at app/templates/components/bs-accordion/item/body.hbs
app/components/bs-accordion/item/title.js: Could not find template at app/templates/components/bs-accordion/item/title.hbs
app/components/bs-dropdown/menu/divider.js: Could not find template at app/templates/components/bs-dropdown/menu/divider.hbs
app/components/bs-dropdown/menu/item.js: Could not find template at app/templates/components/bs-dropdown/menu/item.hbs
app/components/bs-dropdown/menu/link-to.js: Could not find template at app/templates/components/bs-dropdown/menu/link-to.hbs
app/components/bs-form/element/control.js: Could not find template at app/templates/components/bs-form/element/control.hbs
app/components/bs-form/element/errors.js: Could not find template at app/templates/components/bs-form/element/errors.hbs
app/components/bs-form/element/feedback-icon.js: Could not find template at app/templates/components/bs-form/element/feedback-icon.hbs
app/components/bs-form/element/help-text.js: Could not find template at app/templates/components/bs-form/element/help-text.hbs
app/components/bs-form/element/label.js: Could not find template at app/templates/components/bs-form/element/label.hbs
app/components/bs-modal/header/close.js: Could not find template at app/templates/components/bs-modal/header/close.hbs
app/components/bs-modal/header/title.js: Could not find template at app/templates/components/bs-modal/header/title.hbs
app/components/bs-form/element/control/checkbox.js: Could not find template at app/templates/components/bs-form/element/control/checkbox.hbs
app/components/bs-form/element/control/input.js: Could not find template at app/templates/components/bs-form/element/control/input.hbs
app/components/bs-form/element/control/radio.js: Could not find template at app/templates/components/bs-form/element/control/radio.hbs
app/components/bs-form/element/control/textarea.js: Could not find template at app/templates/components/bs-form/element/control/textarea.hbs
app/components/bs-form/element/layout/horizontal.js: Could not find template at app/templates/components/bs-form/element/layout/horizontal.hbs
app/components/bs-form/element/layout/inline.js: Could not find template at app/templates/components/bs-form/element/layout/inline.hbs
app/components/bs-form/element/layout/vertical.js: Could not find template at app/templates/components/bs-form/element/layout/vertical.hbs
app/components/bs-form/element/layout/horizontal/checkbox.js: Could not find template at app/templates/components/bs-form/element/layout/horizontal/checkbox.hbs
app/components/bs-form/element/layout/inline/checkbox.js: Could not find template at app/templates/components/bs-form/element/layout/inline/checkbox.hbs
app/components/bs-form/element/layout/vertical/checkbox.js: Could not find template at app/templates/components/bs-form/element/layout/vertical/checkbox.hbs
addon/components/base/bs-accordion.js: <div>...</div>
addon/components/base/bs-alert.js: Unsupported non-boolean `@classNameBindings` value: alert
addon/components/base/bs-button-group.js: Unsupported non-boolean `@classNameBindings` value: sizeClass
addon/components/base/bs-button.js: Unsupported non-boolean `@classNameBindings` value: active
addon/components/base/bs-carousel.js: Unsupported component type.
addon/components/base/bs-collapse.js: Could not find template at addon/templates/components/base/bs-collapse.hbs
addon/components/base/bs-contextual-help.js: Could not find template at addon/templates/components/base/bs-contextual-help.hbs
addon/components/base/bs-dropdown.js: Unsupported non-boolean `@classNameBindings` value: containerClass
addon/components/base/bs-form.js: Unsupported non-boolean `@classNameBindings` value: layoutClass
addon/components/base/bs-modal-simple.js: Unsupported component type.
addon/components/base/bs-modal.js: skipping tagless component
addon/components/base/bs-nav.js: Unsupported non-boolean `@classNameBindings` value: typeClass
addon/components/base/bs-navbar.js: Could not find template at addon/templates/components/base/bs-navbar.hbs
addon/components/base/bs-popover.js: Unsupported component type.
addon/components/base/bs-progress.js: <div>...</div>
addon/components/base/bs-tab.js: Unsupported component type.
addon/components/base/bs-tooltip.js: Unsupported component type.
addon/components/bs4/bs-accordion.js: Could not find template at addon/templates/components/bs4/bs-accordion.hbs
addon/components/bs4/bs-alert.js: Could not find template at addon/templates/components/bs4/bs-alert.hbs
addon/components/bs4/bs-button-group.js: Could not find template at addon/templates/components/bs4/bs-button-group.hbs
addon/components/bs4/bs-button.js: Could not find template at addon/templates/components/bs4/bs-button.hbs
addon/components/bs4/bs-carousel.js: Could not find template at addon/templates/components/bs4/bs-carousel.hbs
addon/components/bs4/bs-collapse.js: Could not find template at addon/templates/components/bs4/bs-collapse.hbs
addon/components/bs4/bs-dropdown.js: Could not find template at addon/templates/components/bs4/bs-dropdown.hbs
addon/components/bs4/bs-form.js: Could not find template at addon/templates/components/bs4/bs-form.hbs
addon/components/bs4/bs-modal-simple.js: Could not find template at addon/templates/components/bs4/bs-modal-simple.hbs
addon/components/bs4/bs-modal.js: Could not find template at addon/templates/components/bs4/bs-modal.hbs
addon/components/bs4/bs-nav.js: Could not find template at addon/templates/components/bs4/bs-nav.hbs
addon/components/bs4/bs-navbar.js: Unsupported component type.
addon/components/bs4/bs-popover.js: Could not find template at addon/templates/components/bs4/bs-popover.hbs
addon/components/bs4/bs-progress.js: Could not find template at addon/templates/components/bs4/bs-progress.hbs
addon/components/bs4/bs-tab.js: Could not find template at addon/templates/components/bs4/bs-tab.hbs
addon/components/bs4/bs-tooltip.js: Could not find template at addon/templates/components/bs4/bs-tooltip.hbs
addon/components/bs3/bs-accordion.js: Could not find template at addon/templates/components/bs3/bs-accordion.hbs
addon/components/bs3/bs-alert.js: Could not find template at addon/templates/components/bs3/bs-alert.hbs
addon/components/bs3/bs-button-group.js: Could not find template at addon/templates/components/bs3/bs-button-group.hbs
addon/components/bs3/bs-button.js: Could not find template at addon/templates/components/bs3/bs-button.hbs
addon/components/bs3/bs-carousel.js: Could not find template at addon/templates/components/bs3/bs-carousel.hbs
addon/components/bs3/bs-collapse.js: Could not find template at addon/templates/components/bs3/bs-collapse.hbs
addon/components/bs3/bs-dropdown.js: Could not find template at addon/templates/components/bs3/bs-dropdown.hbs
addon/components/bs3/bs-form.js: Could not find template at addon/templates/components/bs3/bs-form.hbs
addon/components/bs3/bs-modal-simple.js: Could not find template at addon/templates/components/bs3/bs-modal-simple.hbs
addon/components/bs3/bs-modal.js: Could not find template at addon/templates/components/bs3/bs-modal.hbs
addon/components/bs3/bs-nav.js: Could not find template at addon/templates/components/bs3/bs-nav.hbs
addon/components/bs3/bs-navbar.js: Unsupported component type.
addon/components/bs3/bs-popover.js: Could not find template at addon/templates/components/bs3/bs-popover.hbs
addon/components/bs3/bs-progress.js: Could not find template at addon/templates/components/bs3/bs-progress.hbs
addon/components/bs3/bs-tab.js: Could not find template at addon/templates/components/bs3/bs-tab.hbs
addon/components/bs3/bs-tooltip.js: Could not find template at addon/templates/components/bs3/bs-tooltip.hbs
addon/components/base/bs-accordion/item.js: Unsupported non-boolean `@classNameBindings` value: disabled
addon/components/base/bs-button-group/button.js: Could not find template at addon/templates/components/base/bs-button-group/button.hbs
addon/components/base/bs-carousel/slide.js: Unsupported component type.
addon/components/base/bs-contextual-help/element.js: Could not find template at addon/templates/components/base/bs-contextual-help/element.hbs
addon/components/base/bs-dropdown/button.js: Could not find template at addon/templates/components/base/bs-dropdown/button.hbs
addon/components/base/bs-dropdown/menu.js: Could not find template at addon/templates/components/base/bs-dropdown/menu.hbs
addon/components/base/bs-dropdown/toggle.js: Could not find template at addon/templates/components/base/bs-dropdown/toggle.hbs
addon/components/base/bs-form/element.js: Unsupported component type.
addon/components/base/bs-form/group.js: <div>...</div>
addon/components/base/bs-modal/body.js: <div>...</div>
addon/components/base/bs-modal/dialog.js: Using `this.element` is not supported in tagless components
addon/components/base/bs-modal/footer.js: <form>...</form>
addon/components/base/bs-modal/header.js: Could not find template at addon/templates/components/base/bs-modal/header.hbs
addon/components/base/bs-nav/item.js: Unsupported component type.
addon/components/base/bs-nav/link-to.js: Could not find template at addon/templates/components/base/bs-nav/link-to.hbs
addon/components/base/bs-navbar/content.js: Unsupported component type.
addon/components/base/bs-navbar/link-to.js: Could not find template at addon/templates/components/base/bs-navbar/link-to.hbs
addon/components/base/bs-navbar/nav.js: Could not find template at addon/templates/components/base/bs-navbar/nav.hbs
addon/components/base/bs-navbar/toggle.js: Could not find template at addon/templates/components/base/bs-navbar/toggle.hbs
addon/components/base/bs-popover/element.js: Unsupported component type.
addon/components/base/bs-progress/bar.js: Using `this.element` is not supported in tagless components
addon/components/base/bs-tab/pane.js: Unsupported component type.
addon/components/base/bs-tooltip/element.js: Unsupported component type.
addon/components/bs4/bs-accordion/item.js: Could not find template at addon/templates/components/bs4/bs-accordion/item.hbs
addon/components/bs4/bs-button-group/button.js: Could not find template at addon/templates/components/bs4/bs-button-group/button.hbs
addon/components/bs4/bs-carousel/slide.js: Could not find template at addon/templates/components/bs4/bs-carousel/slide.hbs
addon/components/bs4/bs-dropdown/button.js: Could not find template at addon/templates/components/bs4/bs-dropdown/button.hbs
addon/components/bs4/bs-dropdown/menu.js: Unsupported component type.
addon/components/bs4/bs-dropdown/toggle.js: Could not find template at addon/templates/components/bs4/bs-dropdown/toggle.hbs
addon/components/bs4/bs-form/element.js: Could not find template at addon/templates/components/bs4/bs-form/element.hbs
addon/components/bs4/bs-form/group.js: Could not find template at addon/templates/components/bs4/bs-form/group.hbs
addon/components/bs4/bs-modal/body.js: Could not find template at addon/templates/components/bs4/bs-modal/body.hbs
addon/components/bs4/bs-modal/dialog.js: Could not find template at addon/templates/components/bs4/bs-modal/dialog.hbs
addon/components/bs4/bs-modal/footer.js: Could not find template at addon/templates/components/bs4/bs-modal/footer.hbs
addon/components/bs4/bs-modal/header.js: Unsupported component type.
addon/components/bs4/bs-nav/item.js: Could not find template at addon/templates/components/bs4/bs-nav/item.hbs
addon/components/bs4/bs-nav/link-to.js: Could not find template at addon/templates/components/bs4/bs-nav/link-to.hbs
addon/components/bs4/bs-navbar/content.js: Could not find template at addon/templates/components/bs4/bs-navbar/content.hbs
addon/components/bs4/bs-navbar/link-to.js: Could not find template at addon/templates/components/bs4/bs-navbar/link-to.hbs
addon/components/bs4/bs-navbar/nav.js: Could not find template at addon/templates/components/bs4/bs-navbar/nav.hbs
addon/components/bs4/bs-navbar/toggle.js: Unsupported component type.
addon/components/bs4/bs-popover/element.js: Could not find template at addon/templates/components/bs4/bs-popover/element.hbs
addon/components/bs4/bs-progress/bar.js: Could not find template at addon/templates/components/bs4/bs-progress/bar.hbs
addon/components/bs4/bs-tab/pane.js: Could not find template at addon/templates/components/bs4/bs-tab/pane.hbs
addon/components/bs4/bs-tooltip/element.js: Could not find template at addon/templates/components/bs4/bs-tooltip/element.hbs
addon/components/bs3/bs-accordion/item.js: Could not find template at addon/templates/components/bs3/bs-accordion/item.hbs
addon/components/bs3/bs-button-group/button.js: Could not find template at addon/templates/components/bs3/bs-button-group/button.hbs
addon/components/bs3/bs-carousel/slide.js: Could not find template at addon/templates/components/bs3/bs-carousel/slide.hbs
addon/components/bs3/bs-dropdown/button.js: Could not find template at addon/templates/components/bs3/bs-dropdown/button.hbs
addon/components/bs3/bs-dropdown/menu.js: Unsupported component type.
addon/components/bs3/bs-dropdown/toggle.js: Could not find template at addon/templates/components/bs3/bs-dropdown/toggle.hbs
addon/components/bs3/bs-form/element.js: Could not find template at addon/templates/components/bs3/bs-form/element.hbs
addon/components/bs3/bs-form/group.js: Could not find template at addon/templates/components/bs3/bs-form/group.hbs
addon/components/bs3/bs-modal/body.js: Could not find template at addon/templates/components/bs3/bs-modal/body.hbs
addon/components/bs3/bs-modal/dialog.js: Could not find template at addon/templates/components/bs3/bs-modal/dialog.hbs
addon/components/bs3/bs-modal/footer.js: Could not find template at addon/templates/components/bs3/bs-modal/footer.hbs
addon/components/bs3/bs-modal/header.js: Unsupported component type.
addon/components/bs3/bs-nav/item.js: Could not find template at addon/templates/components/bs3/bs-nav/item.hbs
addon/components/bs3/bs-nav/link-to.js: Could not find template at addon/templates/components/bs3/bs-nav/link-to.hbs
addon/components/bs3/bs-navbar/content.js: Could not find template at addon/templates/components/bs3/bs-navbar/content.hbs
addon/components/bs3/bs-navbar/link-to.js: Could not find template at addon/templates/components/bs3/bs-navbar/link-to.hbs
addon/components/bs3/bs-navbar/nav.js: Could not find template at addon/templates/components/bs3/bs-navbar/nav.hbs
addon/components/bs3/bs-navbar/toggle.js: Unsupported component type.
addon/components/bs3/bs-popover/element.js: Could not find template at addon/templates/components/bs3/bs-popover/element.hbs
addon/components/bs3/bs-progress/bar.js: Could not find template at addon/templates/components/bs3/bs-progress/bar.hbs
addon/components/bs3/bs-tab/pane.js: Could not find template at addon/templates/components/bs3/bs-tab/pane.hbs
addon/components/bs3/bs-tooltip/element.js: Could not find template at addon/templates/components/bs3/bs-tooltip/element.hbs
addon/components/base/bs-accordion/item/body.js: Could not find template at addon/templates/components/base/bs-accordion/item/body.hbs
addon/components/base/bs-accordion/item/title.js: Could not find template at addon/templates/components/base/bs-accordion/item/title.hbs
addon/components/base/bs-dropdown/menu/divider.js: Could not find template at addon/templates/components/base/bs-dropdown/menu/divider.hbs
addon/components/base/bs-dropdown/menu/item.js: Could not find template at addon/templates/components/base/bs-dropdown/menu/item.hbs
addon/components/base/bs-dropdown/menu/link-to.js: Could not find template at addon/templates/components/base/bs-dropdown/menu/link-to.hbs
addon/components/base/bs-form/element/control.js: Could not find template at addon/templates/components/base/bs-form/element/control.hbs
addon/components/base/bs-form/element/errors.js: skipping tagless component
addon/components/base/bs-form/element/feedback-icon.js: skipping tagless component
addon/components/base/bs-form/element/help-text.js: <div>...</div>
addon/components/base/bs-form/element/label.js: Could not find template at addon/templates/components/base/bs-form/element/label.hbs
addon/components/base/bs-form/element/layout.js: Could not find template at addon/templates/components/base/bs-form/element/layout.hbs
addon/components/base/bs-modal/header/close.js: <button>...</button>
addon/components/base/bs-modal/header/title.js: <h4>...</h4>
addon/components/bs4/bs-accordion/item/body.js: Could not find template at addon/templates/components/bs4/bs-accordion/item/body.hbs
addon/components/bs4/bs-accordion/item/title.js: Could not find template at addon/templates/components/bs4/bs-accordion/item/title.hbs
addon/components/bs4/bs-dropdown/menu/divider.js: Could not find template at addon/templates/components/bs4/bs-dropdown/menu/divider.hbs
addon/components/bs4/bs-dropdown/menu/item.js: Could not find template at addon/templates/components/bs4/bs-dropdown/menu/item.hbs
addon/components/bs4/bs-dropdown/menu/link-to.js: Could not find template at addon/templates/components/bs4/bs-dropdown/menu/link-to.hbs
addon/components/bs4/bs-form/element/control.js: Could not find template at addon/templates/components/bs4/bs-form/element/control.hbs
addon/components/bs4/bs-form/element/errors.js: Could not find template at addon/templates/components/bs4/bs-form/element/errors.hbs
addon/components/bs4/bs-form/element/feedback-icon.js: Could not find template at addon/templates/components/bs4/bs-form/element/feedback-icon.hbs
addon/components/bs4/bs-form/element/help-text.js: Could not find template at addon/templates/components/bs4/bs-form/element/help-text.hbs
addon/components/bs4/bs-form/element/label.js: Unsupported component type.
addon/components/bs4/bs-form/element/layout.js: Could not find template at addon/templates/components/bs4/bs-form/element/layout.hbs
addon/components/bs4/bs-modal/header/close.js: Could not find template at addon/templates/components/bs4/bs-modal/header/close.hbs
addon/components/bs4/bs-modal/header/title.js: Could not find template at addon/templates/components/bs4/bs-modal/header/title.hbs
addon/components/bs3/bs-accordion/item/body.js: Could not find template at addon/templates/components/bs3/bs-accordion/item/body.hbs
addon/components/bs3/bs-accordion/item/title.js: Could not find template at addon/templates/components/bs3/bs-accordion/item/title.hbs
addon/components/bs3/bs-dropdown/menu/divider.js: Could not find template at addon/templates/components/bs3/bs-dropdown/menu/divider.hbs
addon/components/bs3/bs-dropdown/menu/item.js: Could not find template at addon/templates/components/bs3/bs-dropdown/menu/item.hbs
addon/components/bs3/bs-dropdown/menu/link-to.js: Could not find template at addon/templates/components/bs3/bs-dropdown/menu/link-to.hbs
addon/components/bs3/bs-form/element/control.js: Could not find template at addon/templates/components/bs3/bs-form/element/control.hbs
addon/components/bs3/bs-form/element/errors.js: Could not find template at addon/templates/components/bs3/bs-form/element/errors.hbs
addon/components/bs3/bs-form/element/feedback-icon.js: Could not find template at addon/templates/components/bs3/bs-form/element/feedback-icon.hbs
addon/components/bs3/bs-form/element/help-text.js: Could not find template at addon/templates/components/bs3/bs-form/element/help-text.hbs
addon/components/bs3/bs-form/element/label.js: Unsupported component type.
addon/components/bs3/bs-form/element/layout.js: Could not find template at addon/templates/components/bs3/bs-form/element/layout.hbs
addon/components/bs3/bs-modal/header/close.js: Could not find template at addon/templates/components/bs3/bs-modal/header/close.hbs
addon/components/bs3/bs-modal/header/title.js: Could not find template at addon/templates/components/bs3/bs-modal/header/title.hbs
addon/components/base/bs-form/element/control/checkbox.js: Could not find template at addon/templates/components/base/bs-form/element/control/checkbox.hbs
addon/components/base/bs-form/element/control/input.js: Could not find template at addon/templates/components/base/bs-form/element/control/input.hbs
addon/components/base/bs-form/element/control/radio.js: Could not find template at addon/templates/components/base/bs-form/element/control/radio.hbs
addon/components/base/bs-form/element/control/textarea.js: Could not find template at addon/templates/components/base/bs-form/element/control/textarea.hbs
addon/components/base/bs-form/element/layout/horizontal.js: Could not find template at addon/templates/components/base/bs-form/element/layout/horizontal.hbs
addon/components/base/bs-form/element/layout/inline.js: Could not find template at addon/templates/components/base/bs-form/element/layout/inline.hbs
addon/components/base/bs-form/element/layout/vertical.js: Could not find template at addon/templates/components/base/bs-form/element/layout/vertical.hbs
addon/components/bs4/bs-form/element/control/checkbox.js: Could not find template at addon/templates/components/bs4/bs-form/element/control/checkbox.hbs
addon/components/bs4/bs-form/element/control/input.js: Could not find template at addon/templates/components/bs4/bs-form/element/control/input.hbs
addon/components/bs4/bs-form/element/control/radio.js: Unsupported component type.
addon/components/bs4/bs-form/element/control/textarea.js: Could not find template at addon/templates/components/bs4/bs-form/element/control/textarea.hbs
addon/components/bs4/bs-form/element/layout/horizontal.js: Unsupported component type.
addon/components/bs4/bs-form/element/layout/inline.js: Could not find template at addon/templates/components/bs4/bs-form/element/layout/inline.hbs
addon/components/bs4/bs-form/element/layout/vertical.js: Unsupported component type.
addon/components/bs3/bs-form/element/control/checkbox.js: Could not find template at addon/templates/components/bs3/bs-form/element/control/checkbox.hbs
addon/components/bs3/bs-form/element/control/input.js: Could not find template at addon/templates/components/bs3/bs-form/element/control/input.hbs
addon/components/bs3/bs-form/element/control/radio.js: Unsupported component type.
addon/components/bs3/bs-form/element/control/textarea.js: Could not find template at addon/templates/components/bs3/bs-form/element/control/textarea.hbs
addon/components/bs3/bs-form/element/layout/horizontal.js: Unsupported component type.
addon/components/bs3/bs-form/element/layout/inline.js: Could not find template at addon/templates/components/bs3/bs-form/element/layout/inline.hbs
addon/components/bs3/bs-form/element/layout/vertical.js: Unsupported component type.
addon/components/base/bs-form/element/layout/inline/checkbox.js: Could not find template at addon/templates/components/base/bs-form/element/layout/inline/checkbox.hbs
addon/components/base/bs-form/element/layout/vertical/checkbox.js: Could not find template at addon/templates/components/base/bs-form/element/layout/vertical/checkbox.hbs
addon/components/bs4/bs-form/element/layout/horizontal/checkbox.js: Unsupported component type.
addon/components/bs4/bs-form/element/layout/inline/checkbox.js: Could not find template at addon/templates/components/bs4/bs-form/element/layout/inline/checkbox.hbs
addon/components/bs4/bs-form/element/layout/vertical/checkbox.js: Unsupported component type.
addon/components/bs3/bs-form/element/layout/horizontal/checkbox.js: Unsupported component type.
addon/components/bs3/bs-form/element/layout/inline/checkbox.js: Could not find template at addon/templates/components/bs3/bs-form/element/layout/inline/checkbox.hbs
addon/components/bs3/bs-form/element/layout/vertical/checkbox.js: Unsupported component type.
```

### CODE REVIEW

File: addon/components/base/bs-accordion.js [hunk 1]
```diff
@@ -36,7 +36,7 @@ import defaultValue from 'ember-bootstrap/utils/default-decorator';
   @extends Ember.Component
   @public
 */
-@templateLayout(layout)
```
Comment (simonihmig) [inline addon/components/base/bs-accordion.js]: Hm removing that decorator seems like a bug of the codemod!

**Comment by jelhan** [comment]
I reviewed the log of the codemod to get a better understanding why most of the components are skipped. I think it's due to two reasons:

1. Non-boolean `@classNameBindings` values aren't supported yet.
2. Components extending from other components aren't supported yet.

The first one affects seven components:

```
addon/components/base/bs-alert.js: Unsupported non-boolean `@classNameBindings` value: alert
addon/components/base/bs-button-group.js: Unsupported non-boolean `@classNameBindings` value: sizeClass
addon/components/base/bs-button.js: Unsupported non-boolean `@classNameBindings` value: active
addon/components/base/bs-dropdown.js: Unsupported non-boolean `@classNameBindings` value: containerClass
addon/components/base/bs-form.js: Unsupported non-boolean `@classNameBindings` value: layoutClass
addon/components/base/bs-nav.js: Unsupported non-boolean `@classNameBindings` value: typeClass
addon/components/base/bs-accordion/item.js: Unsupported non-boolean `@classNameBindings` value: disabled
```

A strategy to add support is discussed in https://github.com/ember-codemods/tagless-ember-components-codemod/issues/2. But not sure if we need such a generic solution. We could be sure that it's not ambitious for our code base. We could get the type of the value from it's default values and could doublecheck that there isn't a type conversion going on. Additionally we could check that the strings aren't `camelCase`. So this might be another quick win.

I think all other components are skipped cause they extend from another component or are used as a base component for others. This is a complex topic. It would not only mean that the codemod need to parse the parent classes until reaching `@ember/component` but also that it would need to understand at which level of the prototype chain the new template needs to be created. I fear that this would add too much complexity to the codemod.

**Comment by simonihmig** [comment]
> Non-boolean @classNameBindings values aren't supported yet.

I think we can convert these by hand, if there are just 7 components!?

> Components extending from other components aren't supported yet.
> ...
> I fear that this would add too much complexity to the codemod.

Agree. What we could do is to patch the codemod for our purposes and just ignore what they extend from, as we will run it on things that we know are components.

**Comment by jelhan** [comment]
> > Non-boolean @classNameBindings values aren't supported yet.
> 
> I think we can convert these by hand, if there are just 7 components!?

I fear my wording wasn't correct. It affects at least 7 components. This does not include components that are extended from other ones as these check is done first.

> > Components extending from other components aren't supported yet.
> > ...
> > I fear that this would add too much complexity to the codemod.
> 
> Agree. What we could do is to patch the codemod for our purposes and just ignore what they extend from, as we will run it on things that we know are components.

This may allow us to process the files but will likely end up in wrong results. Decorators that affect the wrapping element but are only applied to the base class would be ignored. Additionally this won't solve the issue of shared components.

Lets take `<BsAccordion>` as an example. This is the current code:

```js
//  simplified ember-bootstrap/addon/components/base/bs-accordion.js 
@templateLayout(layout)
export default class Accordion extends Component {
  @defaultValue
  ariaRole = 'tablist';
}
```

https://github.com/kaliber5/ember-bootstrap/blob/ad2c5fbfcf0e1fce91a428a440739044e39ba742/addon/templates/components/common/bs-accordion.hbs#L1-L6

https://github.com/kaliber5/ember-bootstrap/blob/ad2c5fbfcf0e1fce91a428a440739044e39ba742/addon/components/bs3/bs-accordion.js#L1-L5

https://github.com/kaliber5/ember-bootstrap/blob/ad2c5fbfcf0e1fce91a428a440739044e39ba742/addon/components/bs4/bs-accordion.js#L1-L5

The expected output would be:

1. Remove `@templateLayout(layout)` decorator and `ariaRole` property from `ember-bootstrap/addon/components/base/bs-accordion.js`
2. Delete `ember-bootstrap/addon/templates/components/common/bs-accordion.hbs` template.
3. Create a template at `ember-bootstrap/addon/templates/components/bs3/bs-accordion.hbs` with the following content:
   ```hbs
   <div
     role="tablist" {{! ariaRole property from base class }}
     class="accordion" {{! classNames decorator from bs3 specific class }}
   >
    {{! inline template from base class }}
   </div>
   ```
4. Create a similar template at `ember-bootstrap/addon/templates/components/bs4/bs-accordion.hbs` but using the `@classNames` decorator value from bs4-specific class.
5. Add `layout` decorators to bs3 and bs4 specific classes.

**Comment by simonihmig** [comment]
> Remove @templateLayout(layout)

Why that?

> Agree. What we could do is to patch the codemod for our purposes and just ignore what they extend from, as we will run it on things that we know are components.

OK, I see that does not make sense. There are some `base` components that extend from others (tooltip, popover, modal-simple), bit  those are tag-less anyway.

Maybe it would help to also move the `bs<x>` templates into `addon/templates/components/base` to work around the `Could not find template` error we see so often. Like
* mv addon/templates/components/common addon/templates/components/base
* run codemod
* revert changes to JS classes
* mv addon/templates/components/base addon/templates/components/common
* mv addon/templates/components/bs3 addon/templates/components/base
* run codemod
* revert changes to JS classes
* mv addon/templates/components/base addon/templates/components/b3
* mv addon/templates/components/bs4 addon/templates/components/base
* run codemod
* *keep* changes to JS classes
* mv addon/templates/components/base addon/templates/components/b4

That way we at least can convert components that have BS-specific templates. 

> We could get the type of the value from it's default values and could doublecheck that there isn't a type conversion going on. 

Ok, I could try this to add this to the codemod, without upstreaming that, just to make it work for our very specific cases. Maybe just assume it's a non-boolean (if no default value), and warn about it, so we can double check manually?

File: addon/components/base/bs-accordion.js (hunk 1 above)
Comment (simonihmig) [reply addon/components/base/bs-accordion.js]: Fixed that, in the codemod PR!

**Pull request closed without merging**